packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -153,7 +155,10 @@ const App = (props) => {
         <Box className="sf-app" {...styles.container}>
             {/* For pwa-kit v2, we need to use this getToken prop to make sure it can work properly
                 with Storefront Preview feature*/}
-            <StorefrontPreview getToken={() => window.localStorage.getItem('token')}>
+            <StorefrontPreview
+                getToken={() => window.localStorage.getItem('token')}
+                apiClients={apiClients}
+            >
```

Comment:
To consume the update, existing projects will need to do the following:
- Set a new `apiClients` prop for the `StorefrontPreview` component
- Update their dependencies for: `pwa-kit-react-sdk` and `commerce-sdk-isomorphic`

Reply:
@vmarta if you had to give me a jsdoc style description of that the `apiClient` props is, what would it be?

Reply:
@bendvc there's already a jsdoc comment describing that `apiClient` prop (see the other file "packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/index.jsx"). Did you mean that that is not descriptive enough?
`* @param {Object} apiClients - CommerceAPI instance`

Reply:
Yes... the comment doesn't really let the end user know what it does or why you can pass the api clients to it. 

The reason I'm asking this question is because I think it might be hard to describe why we are passing that object in in 1 or 2 sentences which might be a code smell that you are doing something wrong. 

Looking at the pattern you have, it doesn't really follow typical patterns for components. Usually you pass in a prop and it either will be used as options or a conditional or rendering, or used directly to display data that you are passing in. 

The pattern you have here is you pass in an object to the component and when that component mounts you modify the object, and that prop isn't used again.

I think there is probably a better pattern that we can create that won't require use to add this apiClients props onto the component, but maybe something more generic.